docs/architecture/adr-056-proving-amnesia-attacks.md

```diff
@@ -0,0 +1,120 @@
+
+### Amnesia Attack
+
+The currently only known form of global evidence stems from [flip flopping](https://github.com/tendermint/spec/blob/master/spec/consensus/light-client/accountability.md#flip-flopping) attacks. The schematic below explains one scenario where an amnesia attack, a form of flip flopping, can occur such that two sets of honest nodes, C1 and C2, commit different blocks.
+
+![](../imgs/tm-amnesia-attack.png)
+
+1. C1 and F send PREVOTE messages for block A.
+2. C1 sends PRECOMMIT for round 1 for block A.
+3. A new round is started, C2 and F send PREVOTE messages for a different block B.
+4. C2 and F then send PRECOMMIT messages for block B.
+5. F breaks the lock and goes back and sends PRECOMMIT messages in round 1 for block A.
+
+
+This creates a fork on the main chain.  Back to the past, another form of flip flopping, creates a light fork (capable of fooling those not involved in consensus), in a similar way, with F taking the precommits from C1 and forging a commit from them.
+
+## Decision
+
+As the distinction between these two attacks (amnesia and back to the past) can only be distinguished by confirming with all validators (to see if it is a full fork or a light fork), for the purpose of simplicity, these attacks will be treated as the same.
+
+Currently, the evidence reactor is used to simply broadcast and store evidence. Instead of perhaps creating a new reactor for the specific task of verifying these attacks, the current evidence reactor will be extended.
+
+The process begins with a light client receiving conflicting headers (in the future this could also be a full node during fast sync), which it sends to a full node to analyse. As part of [evidence handling](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-047-handling-evidence-from-light-client.md), this could be deduced into potential amnesia evidence
+
+```golang
+type PotentialAmnesiaEvidence struct {
+	V1 []*types.Vote
+	V2 []*types.Vote
+
+	timestamp time.Time
+}
+```
+
+*NOTE: Unlike prior evidence types, `PotentialAmnesiaEvidence` and `AmnesiaEvidence` are processed as a batch instead
+ of individually. This will require changes to much of the API.*
+
+ *NOTE: `PotentialAmnesiaEvidence` could be constructed for when 1/3 or less vote in two different rounds but as it is not currently detected nor can it cause a fork, it will be ignored.*
```

Comment:
I'd say more importantly we don't get DoS with potential amnesia evidence which turns out fake

Reply:
btw, how easy right now is to create a 1/3+ amnesia?

Reply:
You mean if I was already going to be slashed that, if I was still a validator I could start purposefully double voting to create as much evidence as possible in order to try crash the system?

Reply:
Yeah, like if I want to spam the network with PotentialAmnesiaEvidence, how easy is to create 2 conflicting headers. Let's say there were 2 rounds (0 and 1). 0 - quorum was not reached (2/3 voted, without +). I (attacker) take 2/3 and forge additional vote (remember I dont even need to be a part of validator set). Now I created a alternative block (light fork).

Reply:
Sorry, this is not really related to this ADR.